config/smarty.config.inc.php

```diff
@@ -177,9 +177,8 @@ function smartyCleanHtml($data)
 function smartyClassname($classname)
 {
     $classname = Tools::replaceAccentedChars(strtolower($classname));
-    $classname = preg_replace('/[^A-Za-z0-9]/', '-', $classname);
-    $classname = preg_replace('/[-]+/', '-', $classname);
-
+    $classname = preg_replace('/[^A-Za-z0-9-]/', "¤", $classname);
+    $classname = preg_replace(array("/[¤]+/", "/[-]{3,}/", "/[-]+$/"), array('-', '-', '') , $classname);
```

Comment:
Alternative with only one replace:
```suggestion
$classname = preg_replace(array('/[^A-Za-z0-9-]+/', '/[-]{3,}/', '/[-]+$/'), array("-", '-', ''), $classname);
```

Funny story: `a@@--test` will yield `a-test` while `a@@-test` will yield `a--test`

Reply:
Well seen ! I just changed. 
I also added the underscore for the elements of the BEM method